app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.kt

```diff
@@ -54,7 +55,8 @@ class AddEditTaskFragment : Fragment() {
         super.onActivityCreated(savedInstanceState)
         setupSnackbar()
         setupNavigation()
-        loadData()
+        this.setupRefreshLayout(viewDataBinding.refreshLayout)
+        viewModel.start(getTaskId())
```

Comment:
In `TaskDetailFragment ` you simply call the code that you've extracted here to getTaskId - I would remove the getTaskId, since I think this is the only place you call it.

Also I think you should be using by navArgs: https://developer.android.com/guide/navigation/navigation-pass-data

Reply:
Done